### PR TITLE
update Nebraska 2024 retirement income and social security exclusions

### LINKED ISSUE #4795: Nebraska 2024 retirement income and social security exclusions
[Reference](https://revenue.nebraska.gov/about/2023-nebraska-legislative-changes)
image

[Bill](https://www.nebraskalegislature.gov/FloorDocs/108/PDF/Slip/LB754.pdf)

### PR DESCRIPTION
**Pull request by yuantian0204**
fixes #4795

### CODE REVIEW

**Comment by yuantian0204** [comment]
I run git pull upstream master and it says it's already up to date

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need changelog

File: policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/ne_agi_subtractions.py [hunk 2]
```diff
@@ -23,4 +23,7 @@ def formula(tax_unit, period, parameters):
             p.social_security.fraction,
         )
         taxable_oasdi = add(tax_unit, period, ["taxable_social_security"])
-        return taxable_oasdi * oasdi_fraction
+        taxable_retirement_pension = add(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/ne_agi_subtractions.py:26]: this only comes into effect in 2024, need a bool to reflect that

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/ne_agi_subtractions.py:26]: which then is used in an if statement

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/social_security/fraction.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/social_security/fraction.yaml:1]: create an `in_effect.yaml` file

File: policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/ne_agi_subtractions.py [hunk 3]
```diff
@@ -23,4 +23,10 @@ def formula(tax_unit, period, parameters):
...
+            tax_unit, period, ["taxable_public_pension_income"]
+        )
+        if period == 2024:
+            return taxable_oasdi * oasdi_fraction + taxable_retirement_pension
+        else:
+            return taxable_oasdi * oasdi_fraction
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/ne_agi_subtractions.py:32]:
```suggestion
        oasdi_fraction = taxable_oasdi * oasdi_fraction
        if p.in_effect:
              taxable_retirement_pension = add(
                   tax_unit, period, ["taxable_public_pension_income"]
                 )
                 return oasdi_fraction + taxable_retirement_pension
        else:
            return oasdi_fraction
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Comment by PavelMakarchuk** [comment]
refactored the subtractions variable to match code standards by adding a list parameter

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    - ne_social_security_subtraction
+    - taxable_public_pension_income
+    
+
+metadata:
+  unit: list
+  period: year
+  label: Nebraska subtractions from federal adjusted gross income
+  reference:
+    - title: Nebraska Revised Statute | chapter 77 | §2716
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml:15]: What sections? I only see 13 and 14 for Social Security

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+description: Nebraska subtracts these sources from adjusted gross income.
+values:
+  2021-01-01:
+    - ne_social_security_subtraction
+  2024-01-01:
+    - ne_social_security_subtraction
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml:6]: Right? Per the screenshot in the issue, and the legal text, the threshold no longer applies. Or if we want to stay closer to the law, keep the fraction but make the threshold infinite.

> (14)(a) For taxable years beginning or deemed to begin on or after January 1, 2021, under the Internal Revenue Code of 1986, as amended, federal adjusted gross income shall be reduced by a percentage of the social security benefits that are received and included in federal adjusted gross income. The pertinent percentage shall be:

> 
> (i) Five percent for taxable years beginning or deemed to begin on or after January 1, 2021, and before January 1, 2022, under the Internal Revenue Code of 1986, as amended;
> 
> (ii) Forty percent for taxable years beginning or deemed to begin on or after January 1, 2022, and before January 1, 2023, under the Internal Revenue Code of 1986, as amended;
> 
> (iii) Sixty percent for taxable years beginning or deemed to begin on or after January 1, 2023, and before January 1, 2024, under the Internal Revenue Code of 1986, as amended; and
> 
> (iv) One hundred percent for taxable years beginning or deemed to begin on or after January 1, 2024, under the Internal Revenue Code of 1986, as amended.
> 
> (b) For purposes of this subsection, social security benefits means benefits received under the federal Social Security Act.
> 
> (c) For taxable years beginning or deemed to begin on or after January 1, 2021, and before January 1, 2024, under the Internal Revenue Code of 1986, as amended, a taxpayer may claim the reduction to federal adjusted gross income allowed under this subsection or the reduction to federal adjusted gross income allowed under subsection (13) of this section, whichever provides the greater reduction.

```suggestion
    - taxable_social_security_benefits
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml (hunk 4 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml:15]: 20:
image

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml:6]: Does the follow not imply 100%:
`One hundred percent for taxable years beginning or deemed to begin on or after January 1, 2024,`

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml:6]: Yes but it also removes the threshold. Please add a test for this.

**Comment by MaxGhenis** [comment]
test failing

**Comment by PavelMakarchuk** [comment]
> test failing

Interesting, we cant use .inf for thresholds due to the uprating:
image

**Review by MaxGhenis — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml (hunk 4 above)
Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml:15]: Please specify the section in the title. No need for a PR now.

**Pull request merged**